sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/implementation/ManagementLocksImpl.java

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.resources.implementation;
+
+import com.azure.core.http.rest.PagedFlux;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.util.CoreUtils;
+import com.azure.resourcemanager.resources.ResourceManager;
+import com.azure.resourcemanager.resources.fluent.models.ManagementLockObjectInner;
+import com.azure.resourcemanager.resources.fluentcore.arm.ResourceUtils;
+import com.azure.resourcemanager.resources.fluentcore.arm.collection.implementation.CreatableResourcesImpl;
+import com.azure.resourcemanager.resources.models.ManagementLock;
+import com.azure.resourcemanager.resources.models.ManagementLocks;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+
+/**
+ *  Implementation for ManagementLocks.
+ */
+public final class ManagementLocksImpl
+    extends CreatableResourcesImpl<ManagementLock, ManagementLockImpl, ManagementLockObjectInner>
+    implements ManagementLocks {
+
+    private final ResourceManager manager;
+
+    public ManagementLocksImpl(final ResourceManager manager) {
+        this.manager = manager;
+    }
+
+    /**
+     * Returns the part of the specified management lock resource ID
+     * representing the resource the lock is associated with.
+     * @param lockId a lock resource ID
+     * @return a resource ID
+     */
+    static String resourceIdFromLockId(String lockId) {
+        String[] lockIdParts = lockIdParts(lockId);
+        if (CoreUtils.isNullOrEmpty(lockIdParts)) {
+            return null;
+        }
+
+        StringBuilder resourceId = new StringBuilder();
+        for (int i = 0; i < lockIdParts.length - 4; i++) {
+            if (!lockIdParts[i].isEmpty()) {
+                resourceId.append("/").append(lockIdParts[i]);
+            }
+        }
+
+        return resourceId.toString();
+    }
+
+    private static String[] lockIdParts(String lockId) {
+        if (CoreUtils.isNullOrEmpty(lockId)) {
+            return new String[0];
+        }
+
+        String[] parts = lockId.split("/");
+        if (parts.length < 4) {
+            // ID too short to be possibly a lock ID
+            return new String[0];
+        }
+
+        if (!parts[parts.length - 2].equalsIgnoreCase("locks")
+                || !parts[parts.length - 3].equalsIgnoreCase("Microsoft.Authorization")
+                || !parts[parts.length - 4].equalsIgnoreCase("providers")) {
+            // Not a lock ID
+            return new String[0];
```

Comment:
Not exactly sure whether we should throw exception.

Reply:
This can be used after a Get/List operation for lockedResourceId. It might be better not throwing here.